monitoring/monitoring_test.go

```diff
@@ -69,10 +69,24 @@ var (
 		"test_gauge",
 		"Testing gauge functionality",
 	)
+
+	testDisabledSum = monitoring.NewSum(
+		"events_disabled_total",
+		"Number of events observed, by name and kind",
+		monitoring.WithLabels(name, kind),
+	)
+
+	testConditionalSum = monitoring.NewSum(
+		"events_conditional_total",
+		"Number of events observed, by name and kind",
+		monitoring.WithLabels(name, kind),
+	)
 )
 
 func init() {
 	monitoring.MustRegister(testSum, hookSum, int64Sum, testDistribution, testGauge)
+	monitoring.RegisterIf(testDisabledSum, func() bool { return false })
```

Comment:
I think this works as well - definitely fine merging. One thing I was also considering is if we wanted to make it conditional in `NewSum` instead of register? Then we could have a `disabledMetric struct` instead of `if disabled { ... }` everywhere.

Reply:
I was thinking about that but was in two minds - but if you prefer, I can change it to NewSum

Reply:
I like the usage (from ux) of RegisterIf but the implementation of NewSum :slightly_smiling_face:. I think this is probably fine

Reply:
Changed RegisterIf to return the registered metric or disabledMetric so that we can use disabledMetric. See if this better.